docs/1-tendermint-overview.md

```diff
@@ -0,0 +1,93 @@
+
+## Consensus algorithm
+
+Tendermint is a partially synchronous BFT consensus protocol. The protocol requires a fixed set of validators that attempt to come to consensus on one value or, more precisely, on a block (a block is a list of transactions) using gossip-based communication. The consensus algorithm proceeds in rounds: in each round there is a validator that proposes a value (proposer) and all the validators then vote during the round on whether to accept the proposed value or move on to the next round (proposers are chosen according to their voting power).
+
+As most of the consensus algorithms, the Tendermint consensus algorithm assumes a correct majority of processes: less than one third of processes are faulty (*f*), therefore the total number of processes *n* is strictly greater than *3f* (*n > 3f*). For simplicity, we can asssume that:
+ 
+    n = 3f + 1
+
+During a round, processes can exchange different messages and we can summarize the execution in two voting steps: Prevote and Precommit steps. A vote can be for a particular value or for a null value.
+
+A correct process decides on a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of Precommit messages for *v* in a round *r*. 
+A correct process sends a Precommit message for a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of Prevote messages for *v* in a round *r*.
+
+Validators wait some time before sending a Prevote with a null value if they do not receive a valid proposal after a certain time and they send a Precommit message with a null value if they do not receive *2f + 1* Prevote messages for a value.
+If a correct process receives more than 2/3 Precommit messages for a null value in the same round, it moves to the next round.
+
+After a decision has been made, processes continue to agree on other values on another height (consensus instance) and they repeat the process described above in order to agree on different transactions.
+
+In order to ensure that processes will eventually come to a decision, there are some constraints that are applied. These rules are used to prevent any malicious attempt to cause more than one block to be committed at a given height. 
+When a validator *p* sends a Precommit for a value *v* at round *r*, we say the *p* is *locked* on *v*. Validators can propose and send Prevote messages for a value they have locked and they can only change the locked value if they receive a more recent proposal with a quorum of Prevote messages. 
+
+These conditions ensure both the safety and liveness properties of the consensus algorithm because a validator cannot send a Precommit message without sufficient evidence and cannot send a Precommit message for a different value at the same time.
```

Comment:
"Evidence" = "justification" in our terminology?

```suggestion
These conditions ensure both the safety and liveness properties of the consensus algorithm because a validator cannot send a Precommit message without sufficient evidence (i.e., justification) and cannot send a Precommit message for a different value in the same round.
```

Reply:
Yes. I changed to justification just to make it uniform across the documentation for better understanding.